dipy/workflows/vis_registeration.py

```diff
@@ -0,0 +1,278 @@
+from __future__ import division, print_function, absolute_import
+from dipy.workflows.workflow import Workflow
+import logging
+import numpy as np
+import nibabel as nib
+from dipy.workflows.core import write_gif
+from dipy.io.image import load_affine_matrix
+from dipy.viz.regtools import overlay_slices
+from dipy.viz import (window, actor)
+
+from dipy.viz.window import snapshot
+
+
+class VisualizeRegisteredImage(Workflow):
+
+    def process_image_data(self, static_img, moved_img):
+
+        """
+        Function for pre-processing the image data. It involves
+        normalizing and copying static, moving images in the
+        red and green channel, respectively.
+
+        type_vis: str
+            The type of visualisation being created. This affects the
+            range of values to be selected from the moved image.
+
+        return: the normalized image data and range of pixels to be
+        selected.
+        """
+
+        static_img = 255 * ((static_img - static_img.min()) /
+                            (static_img.max() - static_img.min()))
+        moved_img = 255 * ((moved_img - moved_img.min()) /
+                           (moved_img.max() - moved_img.min()))
+
+        # Create the color images
+        overlay = np.zeros(shape=(static_img.shape) + (3,), dtype=np.uint8)
+        overlay[..., 0] = static_img
+        overlay[..., 1] = moved_img
+        mean, std = overlay[overlay > 0].mean(), overlay[overlay > 0].std()
+        value_range = (mean - 0.5 * std, mean + 1.5 * std)
+        return overlay, value_range
+
+    def get_row_cols(self, num_slices):
+
+        """
+        Experimetal helper function to get the number
+        of rows and columns for the mosaic.
+
+        num_slices: int
+            The number of slices as obtained from the
+            create_mosaic function.
+        return: the number of rows and columns.
+        """
+        rows, cols = 0, 0
+
+        while True:
+            if num_slices % 5 == 0:
+                break
+            num_slices += 1
+
+        for i in range(5, num_slices):
+
+            if num_slices % i == 0:
+                rows = i
+                cols = num_slices // i
+                break
+
+        return rows, cols
```

Comment:
yeah, I have to find a way to improve this function. The way it is right now it works well for most inputs but it needs more improvement. Not just the conciseness as you suggested but the mathematics itself can be optimized much. I am simply trying to factor a number into its 2 factors so that the number of rows and columns that I get back can be used in the create_mosaic function.

So, if there are 74 slices (for example in an image, I increase it to 75 to make it a multiple of 5). It can be broken down into either 25 *3 or 5 * 15 or 75 * 1 etc. but you will agree that having 75 rows and 1 column or 1 row and 75 column is not the best way to show the slices side by side. Similarly, 25 rows and 3 columns is also not a good option. But, I can use 5 * 15 means 5 rows and 15 columns for visualizing all the slices side by side. 

Arriving at this combination is made difficult by the fact that I can different number of slices in different inputs and what is the most optimal number of rows and columns to create the mosaic?

I need to come up either with a better algorithm or a clever trick to achieve this.

But many thanks for sharing your concerns with me. I really appreciate your help.